pysph/base/stratified_sfc_nnps.pyx

```diff
@@ -69,41 +72,74 @@ cdef class StratifiedSFCNNPS(NNPS):
         self.keys = <uint64_t**> malloc(narrays*sizeof(uint64_t*))
         self.key_to_idx = <int***> malloc(narrays*sizeof(int**))
         self.cell_sizes = <double**> malloc(narrays*sizeof(double*))
+        self.hmax = <double***> malloc(narrays*sizeof(double**))
```

Comment:
This doesn't seem to be free'd.  Can you please check and ensure there are no memory leaks?